L79-cpp-async-response-reader-destruction.md

```diff
@@ -0,0 +1,123 @@
+API changes for `ClientAsyncResponseReaderInterface` destruction
+----
+* Author: jacobsa
+* Approver: ctiller
+* Status: Draft
+* Implemented in: C++
+* Last updated: 2021-05-14
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Currently although
+[`grpc::ClientAsyncResponseReaderInterface`](https://github.com/grpc/grpc/blob/7e14d23ab46e1b0924d6c3e40797ed1d587aee7a/include/grpcpp/impl/codegen/async_unary_call.h#L37-L72)
+has a virtual destructor, `std::default_delete` is specialized such that the
+destructor never actually runs. So the only subclasses that can actually be
+implemeneted without hacks are those like `grpc::ClientAsyncResponseReader`,
+which knows it is always allocated on an arena and doesn't care about its
+destructor running. This proposal provides a plan for fixing this so that other
+subclasses can count on their destructor running and their backing memory being
+freed, as with any other C++ class.
+
+## Background
+
+We currently [specialize](https://github.com/grpc/grpc/blob/7e14d23ab46e1b0924d6c3e40797ed1d587aee7a/include/grpcpp/impl/codegen/async_unary_call.h#L404-L408)
+`std::default_delete` for
+`ClientAsyncResponseReaderInterface` to do nothing. This means that, despite the
+fact that async stub methods return
+`std::unique_ptr<ClientAsyncResponseReaderInterface>`, it's impossible to have a
+subclass of that interface that is allocated on the heap or whose destructor has
+side effects without a hack that manages ownership _outside of_ the `unique_ptr`
+returned by the stub.
+
+This was introduced by
+[`dd36b153`](https://github.com/grpc/grpc/commit/dd36b15315cd691e86a94d4574bd9f3e3a33633f),
+which introduced the specialization for `grpc::ClientAsyncResponseReader`
```

Comment:
nit: I would replace `introduced` with `included` to avoid repetition.

Reply:
Done (except I used `added`).